client/dashboard/domains/dataviews/bulk-actions-progress-notice.tsx

```diff
@@ -0,0 +1,114 @@
+import { bulkDomainUpdateStatusQuery } from '@automattic/api-queries';
+import { useQuery } from '@tanstack/react-query';
+import { __ } from '@wordpress/i18n';
+import { useLayoutEffect, useState } from 'react';
+import { Notice } from '../../components/notice';
+import type { BulkDomainUpdateStatusQueryFnData } from '@automattic/api-core';
+
+const getLastJob = ( data: BulkDomainUpdateStatusQueryFnData | undefined ) => {
+	if ( ! data ) {
+		return undefined;
+	}
+
+	return bulkDomainUpdateStatusQuery()
+		.select?.( data )
+		.allJobs.sort( ( a, b ) => b.created_at - a.created_at )
+		.at( 0 );
+};
+
+export const BulkActionsProgressNotice = () => {
+	const [ lastId, setLastId ] = useState( '' );
+	const [ shouldShowCompleteNotice, setShouldShowCompleteNotice ] = useState( false );
+
+	const { data } = useQuery( {
+		...bulkDomainUpdateStatusQuery(),
+		refetchInterval: ( query ) => {
+			const lastJob = getLastJob( query.state.data );
+
+			return lastJob?.complete ? -1 : 1_000;
+		},
+		meta: { persist: false },
+		staleTime: 0,
+		select: ( data ) => getLastJob( data ),
```

Comment:
Inside of `getLastJob` we also call select on `bulkDomainUpdateStatusQuery`. I'm not sure why that is the case.